eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/ProxyConfigurationTest.java

```diff
@@ -0,0 +1,114 @@
+        Assert.assertFalse(proxyConfiguration.isProxyAddressConfigured());
+
+        proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, null, null, PROXY_PASSWORD);
+        Assert.assertFalse(proxyConfiguration.isProxyAddressConfigured());
+
+        proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, null, PROXY_USERNAME, null);
+        Assert.assertFalse(proxyConfiguration.isProxyAddressConfigured());
+
+        proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, null, null, null);
+        Assert.assertFalse(proxyConfiguration.isProxyAddressConfigured());
+    }
+
+    @Theory
+    public void testHasUserDefinedCredentials(@FromDataPoints("Proxy Configuration Types") ProxyAuthenticationType proxyAuthenticationType) {
+        ProxyConfiguration proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, proxyAddress, PROXY_USERNAME, PROXY_PASSWORD);
+        Assert.assertTrue(proxyConfiguration.hasUserDefinedCredentials());
+
+        proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, proxyAddress, null, PROXY_PASSWORD);
+        Assert.assertFalse(proxyConfiguration.hasUserDefinedCredentials());
+
+        proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, proxyAddress, PROXY_USERNAME, null);
+        Assert.assertFalse(proxyConfiguration.hasUserDefinedCredentials());
+
+        proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, proxyAddress, null, null);
+        Assert.assertFalse(proxyConfiguration.hasUserDefinedCredentials());
+
+        proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, null, PROXY_USERNAME, PROXY_PASSWORD);
+        Assert.assertTrue(proxyConfiguration.hasUserDefinedCredentials());
+
+        proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, null, null, PROXY_PASSWORD);
+        Assert.assertFalse(proxyConfiguration.hasUserDefinedCredentials());
+
+        proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, null, PROXY_USERNAME, null);
+        Assert.assertFalse(proxyConfiguration.hasUserDefinedCredentials());
+
+        proxyConfiguration = new ProxyConfiguration(proxyAuthenticationType, null, null, null);
+        Assert.assertFalse(proxyConfiguration.hasUserDefinedCredentials());
+    }
+
+    private static void validateProxyConfiguration(ProxyConfiguration proxyConfiguration) {
+        String proxyAddressStr = proxyConfiguration.proxyAddress().address().toString();
+        ProxyAuthenticationType authentication = proxyConfiguration.authentication();
+        Assert.assertEquals(HTTP_PROXY, proxyAddressStr);
+        Assert.assertEquals(PROXY_USERNAME, proxyConfiguration.credential().getUserName());
+        Assert.assertEquals(PROXY_PASSWORD, new String(proxyConfiguration.credential().getPassword()));
+        Assert.assertTrue(ProxyAuthenticationType.BASIC.equals(authentication)
```

Comment:
In this case, you should have a parameter in the validateProxyConfiguration with the expected ProxyAuthenticationType rather than trying to assert all of them.

Reply:
ProxyAuthenticatonType is an enum, you should use `Assert.areEquals(ProxyAuthenticationType.BASIC, authentication)`